### PR TITLE
Update unreferenced flag in the summary sent to SPO

### LINKED ISSUE #4057: SPO Driver - Add state in summary sent to server to indicate whether a tree entry is deleted
Once garbage collection runs, it can mark fluid objects (data stores / DDSes) that are no longer referenced. This information needs to be added to that object's summary tree so that the server can know which objects have been collected.

### PR DESCRIPTION
**Pull request by agarwal-navin**
Fixes #4057.

The changes sets the "unreferenced" flag in the summary that is sent to SPO for corresponding entries in the ISummaryTree.

### CODE REVIEW

**Review by arinwt — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -1083,6 +1086,13 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     id,
                     value,
                 };
+
+                // The unreferenced property in a snapshot tree entry indicates whether it is unreferenced. If the
+                // property is not present, the tree entry is considered referenced. If the property is present and is
+                // true (which is the only value it can have), the tree entry is considered unreferenced.
+                if (unreferenced) {
```
Comment (vladsud) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:1093]: Feels like there is no need for it to be conditional statement (i.e. no need for IF) - IMHO more readable and usually more performant (branches are expensive for CPU :))

**Review by vladsud — APPROVED** [review]

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts (hunk 1 above)
Comment (agarwal-navin) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:1093]: I wanted referenced tree entries to not have the property at all. Without the `if` statement, the entries that are referenced will always have the `unreferenced` flag with the value undefined. With the `if` statement, referenced entries will not have the flag at all.

Comment (vladsud) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:1093]: undefined properties are not serialized in JSON (need to double check), so I believe it's not an issue. I.e. I believe it's only state in memory (as there is indeed difference between missing property and undefined property)
Either way is fine. :)

Comment (agarwal-navin) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:1093]: I did not know that :-). I tested it and yeah, JSON does not serialize undefined properties.

**Pull request merged**